### PR TITLE
Add request class for insight service

### PR DESCRIPTION
**Pull request by ricardoperez**
- Add ability to add custom configuration for different AdInsight Reports

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -102,3 +135,9 @@ service.report_body(id) if service.report_ready?(id)
 ```ruby
 result = serive.generate_report({report_type: AuctionInsightData, entity_type: 'Account'})
 ```
+
+### Get Auction Insight Results for Keywords
+
+```ruby
+result = serive.generate_report({report_type: AuctionInsightData, entity_type: 'Account', entity_ids: [1, 2, 3]})
```
Comment (chiting) [inline README.md]: typo with `service`

File: CHANGELOG.txt [hunk 2]
```diff
@@ -1,3 +1,6 @@
+* 4.1.1
```
Comment (chiting) [inline CHANGELOG.txt]: Out of curiosity, what's the versioning policy we're following?

File: README.md (hunk 1 above)
Comment (chiting) [reply README.md]: Or is it meant to be `service`?

File: CHANGELOG.txt (hunk 2 above)
Comment (ricardoperez) [reply CHANGELOG.txt]: Should be Semantic Versioning. I changed the version as I'm not fixing a bug.

File: README.md (hunk 1 above)
Comment (ricardoperez) [reply README.md]: You need to look in the previous line.

**Review by chiting — APPROVED** [review]

File: lib/bing-ads-reporting/insight_service.rb [hunk 3]
```diff
@@ -42,35 +49,10 @@ def report_operation(options)
...
-    end
-
-    def date_range(period)
-      {
-        'EndDate' => scope_data_range(period.to),
-        'StartDate' => scope_data_range(period.from),
-        '@i:type' => 'tns:DateRangeSearchParameter'
-      }
-    end
+      # report_type = options[:report_type]
```
Comment (chiting) [inline lib/bing-ads-reporting/insight_service.rb]: I'd remove these comments if they're not needed

**Review by babakgh — APPROVED** [review]

File: lib/bing-ads-reporting/insight_service.rb (hunk 3 above)
Comment (ricardoperez) [reply lib/bing-ads-reporting/insight_service.rb]: done.

**Pull request merged**